code/modules/antagonists/traitor/objectives/hack_borg.dm

```diff
@@ -0,0 +1,145 @@
+		. += "It can be used to unlock cyborg cover locks and to hack them."
+		if(used)
+			. += "It is already used."
+
+/obj/item/hacking_device/attack()
+	return
+
+/obj/item/hacking_device/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	if(!proximity)
+		return
+	if(iscyborg(target))
+		if(used)
+			to_chat(user, span_warning("The [src] is already used!"))
+			return
+		var/mob/living/silicon/robot/R = target
+		if(R.stat == DEAD)
+			to_chat(user, span_warning("You need to hack a functioning cyborg, not a broken one!"))
+			return
+		if(!R.mind || !R.client)
+			to_chat(user, span_warning("It looks like [R]'s controlling interface doesn't function. Hacking it will make no sense."))
+			return
+		if(!R.opened)//Cover is closed
+			if(R.locked)
+				to_chat(user, span_notice("You hack the cover lock."))
+				R.locked = FALSE
+				if(R.shell) //A warning to Traitors who may not know that emagging AI shells does not slave them.
+					to_chat(user, span_boldwarning("[R] seems to be controlled remotely! Hacking the interface may not work as expected."))
+			else
+				to_chat(user, span_warning("The cover is already unlocked!"))
+			return
+		if(R.wiresexposed)
+			to_chat(user, span_warning("You must unexpose the wires first!"))
+			return
+		if(R.emagged)
+			to_chat(user, span_warning("It looks like [R] is already hacked."))
+			return
+		if(R.connected_ai && R.connected_ai.mind && R.connected_ai.mind.has_antag_datum(/datum/antagonist/malf_ai))
+			to_chat(R, span_danger("ALERT: Foreign software execution prevented."))
+			R.logevent("ALERT: Foreign software execution prevented.")
+			to_chat(R.connected_ai, span_danger("ALERT: Cyborg unit \[[R]\] successfully defended against subversion."))
+			log_silicon("HACK: [key_name(user)] attempted to hack cyborg [key_name(R)], but they were slaved to traitor AI [R.connected_ai].")
+			return
+		if(R.shell) 
+			to_chat(user, span_boldwarning("[R] seems to be controlled remotely! Hacking the interface may not work as expected."))
+			return
+		to_chat(user, span_warning("You sucessfully hack [R]."))
+		used = TRUE
+		R.emag_cooldown = world.time + 150
+		SEND_SIGNAL(src, COMSIG_TRAITOR_BUG_BORG_HACKED)
+		R.SetEmagged(1)
+		R.SetStun(60)
+		R.lawupdate = FALSE
+		R.set_connected_ai(null)
+		message_admins("[ADMIN_LOOKUPFLW(R)] hacked cyborg [ADMIN_LOOKUPFLW(R)].")
+		log_silicon("HACK: [key_name(R)] hacked cyborg [key_name(R)]. Laws overridden.")
+		to_chat(R, span_danger("ALERT: Foreign software detected."))
+		R.logevent("ALERT: Foreign software detected.")
+		sleep(5)
+		to_chat(R, span_danger("Initiating diagnostics..."))
+		sleep(20)
+		to_chat(R, span_danger("LAW SYNCHRONISATION ERROR"))
+		sleep(20)
+		to_chat(R, span_danger("ERRORERRORERROR"))
+		R.add_ion_law(generate_ion_law())
+		sleep(20)
+		var/message = "You have been uploaded with malicious software, that had added a broken law to your lawset. You can't find any memory files about the person who uploaded them to you."
```

Comment:
I'd suggest rewording this to make sure that borgs don't interpret this as forgetting everything they know about the person who interacted with them ("Chemist? We don't have a chemist.") instead of just this one interaction ("Sorry, I don't remember who hacked me.").

Reply:
Ok